### ISSUE TITLE
Tracing and implementing iofuzzing 

### ISSUE THREAD

**Issue by catap** [riboseinc/retrace#141]
Hey,

I think it's also good extend retrace and something likes`memoryfuzzing` but for file IO. Let's say `iofuzzing`.

For rnp we should support:
* `mmap` (where flags isn't `MAP_ANONYMOUS` or `MAP_ANON` in this case it is memoryfuzzing)
* `munmap`
* `open`
* `close`
* `dup`
* `read`
* `write`
* `stat`
* `fdopen`
* `fopen`
* `fclose`
* `fseek`

Usefult but optional for rnp:
* `msync`
* `mprotect`
* `mincore`
* `openat`
* `lseek`
* `umask`
* `fstat`
* `lstat`
* `fstatat`
* `freopen`
* `fileno`
* `funopen`

and doesn't forgot about setup `errno` in some random value :)

also `read`/`write` maybe for network socket and we should ignore it in iofuzzing case

**Comment by ronaldtse** [comment]
Maybe @riataman is interested?

**Comment by riataman** [comment]
Yes! I want to go back to useful feature once I merge this other huge change that I have :)

**Comment by riataman** [comment]
@catap, @ronaldtse: Check this one, I think there's fun to be have with it: https://github.com/riboseinc/retrace/pull/161

**Comment by riataman** [comment]
Merged to paster. Put incompleteio in the config file.

**Comment by ronaldtse** [comment]
Shipped: I/O fuzzing is the `incomplete_io` action (short reads/writes, JSON-configurable, deterministic via the shared seed) — docs/configuration.md, cookbook 10.
